engine/src/main/java/org/terasology/engine/world/chunks/Chunk.java

```diff
@@ -14,5 +21,345 @@
  * For details on dimensions and other chunks characteristics see {@link Chunks}.
  */
 @API
-public interface Chunk extends ManagedChunk, RenderableChunk {
+public interface Chunk extends RenderableChunk {
```

Comment:
I agree that the naming is potentially confusing. I kept it as "RenderableChunk" because there's a lot of overlap with what the `RenderableChunk` interface was before this PR. Also, I'm not sure what a better name would be. Something like "ChunkLikeRenderable" would make sense, but is also pretty verbose.

Reply:
So is a chunk always a renderable chunk :? and in what case would it not be?

Reply:
Yes, a `Chunk` is always a `RenderableChunk`. A `RenderableChunk` would not be a `Chunk` if it doesn't have block data, a position in the chunk grid, or the ability to save & load like an actual chunk.

If your question was "Why would chunks that aren't renderable have been a useful thing to have?", I think the intent of the original interface hierarchy was that in contexts where the rendering information isn't meant to be relevant, one of the superclasses that lacks that feature could be used instead. I ended up having to cast `CoreChunk` to `Chunk` in `SunlightRasterizer` (in CoreWorlds) anyway, so this approach didn't work very well. A non-renderable chunk could, hypothetically, have been used in a headless server, but then you'd need multiple actual implementations, which would be unnecessarily complicated.